src/ucp/tag/eager_rcv.c

```diff
@@ -74,22 +74,24 @@ ucp_eager_handler(void *arg, void *data, size_t length, void *desc,
         }
     }
 
+    rdesc = ucp_recv_desc_get(worker, data, desc, length, hdr_len, flags);
+    if (!rdesc) {
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
     ucs_trace_req("unexp recv %c%c%c tag %"PRIx64" length %zu desc %p",
                   (flags & UCP_RECV_DESC_FLAG_FIRST) ? 'f' : '-',
                   (flags & UCP_RECV_DESC_FLAG_LAST)  ? 'l' : '-',
                   (flags & UCP_RECV_DESC_FLAG_EAGER) ? 'e' : '-',
                   recv_tag, length, rdesc);
 
-    if (data != rdesc + 1) {
-        memcpy(rdesc + 1, data, length);
-    }
-
-    rdesc->length  = length;
-    rdesc->hdr_len = hdr_len;
-    rdesc->flags   = flags;
     ucs_queue_push(&context->tag.unexpected, &rdesc->queue);
 
-    status = UCS_INPROGRESS;
+    if (flags & UCP_RECV_DESC_FLAG_SYNC) {
```

Comment:
so in case of unexpected SYNC, which is received to "data" rather than "desc" (or no AM_FLAG_DESC in the future), the AM callback will return UCS_INPROGRESS (so ACK would not be sent), which is not correct (because desc is not used)
I think need to have 2 indications from `ucp_eager_handler`:
1. whether to release the descriptor
1. whether to send ACK